### PR TITLE
StandaloneMmPkg: Additional CodeQL fixes

### PR DESCRIPTION
**Pull request by makubacki**
## Description

Explicitly forces a NULL check in certain areas.

- [x] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

StandaloneMmPkg build and boot to EFI shell on QemuQ35Pkg.

## Integration Instructions

N/A

Signed-off-by: Michael Kubacki <michael.kubacki@microsoft.com>

### CODE REVIEW

File: StandaloneMmPkg/Core/Dispatcher.c [hunk 1]
```diff
@@ -837,7 +837,12 @@ FvIsBeingProcessed (
   DEBUG ((DEBUG_INFO, "FvIsBeingProcessed - 0x%08x\n", FwVolHeader));
 
   KnownFwVol = AllocatePool (sizeof (KNOWN_FWVOL));
-  ASSERT (KnownFwVol != NULL);
+  // MU_CHANGE [BEGIN] - CodeQL change
+  if (KnownFwVol == NULL) {
+    return;
```
Comment (apop5) [inline StandaloneMmPkg/Core/Dispatcher.c:843]: The assertion removal is intentional? Other file locations moved the assert after the NULL check.

**Review by apop5 — APPROVED** [review]

**Comment by kuqin12** [comment]
I think this change will partially "nullify" the update here: https://github.com/microsoft/mu_feature_mm_supv/pull/76. Do you want to hold off the one PR above and update all override tags in one?

**Review by kuqin12 — APPROVED** [review]

File: StandaloneMmPkg/Library/StandaloneMmHobLib/StandaloneMmHobLib.c [hunk 2]
```diff
@@ -318,6 +318,13 @@ BuildModuleHob (
     );
 
   Hob = CreateHob (EFI_HOB_TYPE_MEMORY_ALLOCATION, sizeof (EFI_HOB_MEMORY_ALLOCATION_MODULE));
+  // MU_CHANGE [BEGIN] - CodeQL change
+  if (Hob == NULL) {
+    ASSERT (Hob != NULL);
```
Comment (kuqin12) [inline StandaloneMmPkg/Library/StandaloneMmHobLib/StandaloneMmHobLib.c:323]: nit: would it be easier to just `ASSERT (FALSE);`?

Comment (makubacki) [reply StandaloneMmPkg/Library/StandaloneMmHobLib/StandaloneMmHobLib.c:323]: The assert is not as obvious in that case.

- `ASSERT (FALSE)` Example:
  - `ASSERT Platforms\QemuQ35Pkg\Library\QemuFwCfgLib\QemuFwCfgLib.c(37): ((BOOLEAN)(0==1))`

- `ASSERT (Pointer != NULL)` Example:
  - `ASSERT Platforms\QemuQ35Pkg\Library\QemuFwCfgLib\QemuFwCfgLib.c(37): Ptr != ((void *) 0)`

File: StandaloneMmPkg/Core/Dispatcher.c (hunk 1 above)
Comment (makubacki) [reply StandaloneMmPkg/Core/Dispatcher.c:843]: The assert removal was unintentional. I'll add it.

**Comment by makubacki** [comment]
> I think this change will partially "nullify" the update here: [microsoft/mu_feature_mm_supv#76](https://github.com/microsoft/mu_feature_mm_supv/pull/76). Do you want to hold off the one PR above and update all override tags in one?

I'll do it again. There's another platform that needs a SUPV update with the new tag before all of these CodeQL changes will go into it.

File: StandaloneMmPkg/Core/Dispatcher.c (hunk 1 above)
Comment (makubacki) [reply StandaloneMmPkg/Core/Dispatcher.c:843]: Updated in PR commit https://github.com/microsoft/mu_basecore/pull/262/commits/e6aeaab787bff72ff473f9b505d7a94f887bddbe

File: StandaloneMmPkg/Library/StandaloneMmHobLib/StandaloneMmHobLib.c (hunk 2 above)
Comment (makubacki) [reply StandaloneMmPkg/Library/StandaloneMmHobLib/StandaloneMmHobLib.c:323]: Going to close for now

**Pull request merged**